app/models/accounting/quickbooks/full_fetcher.rb

```diff
@@ -10,20 +10,56 @@ def initialize(qb_connection = Division.root.qb_connection)
       end
 
       def fetch_all
-        divisions_accounts = Division.clear_all_accounts
+        divisions_accounts = clear_all_accounts!
 
         ::Accounting::Transaction.destroy_all
         ::Accounting::Account.destroy_all
 
         started_fetch_at = Time.zone.now
 
-        ::Accounting::Quickbooks::AccountFetcher.new.fetch
-        ::Accounting::Quickbooks::TransactionFetcher.new.fetch
+        ::Accounting::Quickbooks::AccountFetcher.new(@qb_connection).fetch
+        ::Accounting::Quickbooks::TransactionFetcher.new(@qb_connection).fetch
 
         qb_connection.update_attribute(:last_updated_at, started_fetch_at)
 
-        Division.restore_all_accounts(divisions_accounts)
+        restore_all_accounts!(divisions_accounts)
       end
+
+      private
+
+      # Set all divisions' accounts to nil and return a hash of the QB ids of the removed accounts
+      # by division
+      def clear_all_accounts!
+        divisions_accounts = {}
+        Division.all.each do |d|
```

Comment:
I think we need to keep moving toward doing all QB stuff separately for different divisions. I think this class should accept a Division, not a qb connection, as its only param, and then when we are clearing out accounts like this, it should only do it for the current division. Existing instantiations of this class should explicitly pass the root division for now. I'd rather not have it as a default parameter, that seems odd to me.

Mel, could you make this change without too much trouble?

@linesarefuzzy would you agree?

Reply:
Oh gosh yes I agree. Good catch. This was my branch, so I can fix it.